src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -959,6 +961,87 @@ namespace AppInstaller::CLI::Workflow
             set.Path(absolutePath.wstring());
         }
 
+        // Get Uri zone for a given uri or file path.
+        Settings::ConfigurationAllowedZonesOptions GetUriZone(const std::string& uri)
+        {
+            DWORD dwZone;
+            auto pInternetSecurityManager = winrt::create_instance<IInternetSecurityManager>(CLSID_InternetSecurityManager, CLSCTX_ALL);
+            pInternetSecurityManager->MapUrlToZone(AppInstaller::Utility::ConvertToUTF16(uri).c_str(), &dwZone, 0);
+
+            // Treat all zones higher than untrusted as untrusted
+            if (dwZone > static_cast<DWORD>(Settings::ConfigurationAllowedZonesOptions::UntrustedSites))
+            {
+                return Settings::ConfigurationAllowedZonesOptions::UntrustedSites;
+            }
+
+            return static_cast<Settings::ConfigurationAllowedZonesOptions>(dwZone);
+        }
+
+        // Validate group policy for a given zone.
+        bool IsBlockedByGroupPolicy(Execution::Context& context, const Settings::ConfigurationAllowedZonesOptions zone)
+        {
+            auto configurationPolicies = Settings::GroupPolicies().GetValue<Settings::ValuePolicy::ConfigurationAllowedZones>();
+            if (!configurationPolicies.has_value())
+            {
+                AICLI_LOG(Config, Warning, << "ConfigurationAllowedZones policy is not set");
+                return false;
+            }
+
+            if (configurationPolicies->find(zone) == configurationPolicies->end())
+            {
+                AICLI_LOG(Config, Warning, << "Configuration is not configured in the zone " << zone);
+                return false;
+            }
+
+            auto isAllowed = configurationPolicies->at(zone);
+            if(!isAllowed)
+            {
+                context.Reporter.Error() << "Configuration is disabled for Zone: " << zone << std::endl;
+                return true;
+            }
+
+            AICLI_LOG(Config, Info, << "Configuration is configured in zone " << zone << " with value " << (isAllowed ? "allowed" : "blocked"));
+            return false;
+        }
+
+        // Validate smart screen for a given url.
+        bool IsBlockedBySmartScreen(Execution::Context& context, const std::string& url)

```

Comment:
Should these be in a SmartScreen workflow file, or somewhere other than the configuration workflows? I feel there is opportunity for other URLs to eventually be run through SmartScreen, and segmenting the SmartScreen functionality into its own workflows would make it easier to follow in the future

Reply:
Great feedback! Yes, ultimately this can be beneficial for workflows and other places where we need a consistent logic for dealing with URLs.